src/vs/platform/extensionManagement/common/extensionGalleryService.ts

```diff
@@ -645,6 +649,12 @@ export class ExtensionGalleryService implements IExtensionGalleryService {
 					}
 				});
 			}
+
+			// {{SQL CARBON EDIT}} - filter out extensions that match the excludeFlags options
+			const flags = query.criteria.filter(x => x.filterType === FilterType.ExcludeWithFlags).map(v => v.value ? v.value.toLocaleLowerCase() : undefined);
+			if (flags && flags.length > 0) {
+				filteredExtensions = filteredExtensions.filter(e => !e.flags || flags.find(x => x === e.flags) === undefined);
```

Comment:
Is this going to handle an extension with multiple flags - e.g. `hidden,preview`? We can probably document that as a known limitation since I don't know whether we'd ever actually need such a feature, but it would be good to note any behavior such as that.

Reply:
I was thinking about this, but I suspect that this current extension is going to be the only hidden extension since I doubt many people would want to publish non customer-facing extensions.  I'll document as a limitation and if we find this is a more common use-case we can revisit later.